.gitlab/kernel_version_testing/security_agent.yml

```diff
@@ -0,0 +1,235 @@
+    SECURITY_AGENT_EBPF_BYTECODE: $SECURITY_AGENT_TESTS/pkg/ebpf/bytecode/build
+    AWS_EC2_SSH_KEY_FILE: $CI_PROJECT_DIR/ssh_key
+    TEST_COMPONENT: security-agent
+    ARCHIVE_NAME: dependencies-$ARCH.tar.gz
+    TEST_ARCHIVE_NAME: security-agent-tests-$ARCH.tar.gz
+  artifacts:
+    when: always
+    paths:
+      - $CI_PROJECT_DIR/connector-${ARCH}
+
+upload_security_agent_tests_x64:
+  extends:
+    - .upload_security_agent_tests
+  needs: ["go_deps", "prepare_ebpf_functional_tests_x64", "tests_ebpf_x64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:amd64"]
+  variables:
+    ARCH: x86_64
+    INSTANCE_TYPE: m5d.metal
+
+upload_security_agent_tests_arm64:
+  extends:
+    - .upload_security_agent_tests
+  needs: ["go_deps", "prepare_ebpf_functional_tests_arm64", "tests_ebpf_arm64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:arm64"]
+  variables:
+    ARCH: arm64
+    INSTANCE_TYPE: m6gd.metal
+
+.kernel_matrix_testing_run_security_agent_tests:
+  extends: .kernel_matrix_testing_run_tests
+  stage: kernel_matrix_testing_security_agent
+  rules: !reference [.on_security_agent_changes_or_manual]
+  variables:
+    TEST_COMPONENT: security-agent
+
+kernel_matrix_testing_run_security_agent_tests_x64:
+  extends:
+    - .kernel_matrix_testing_run_security_agent_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:amd64"]
+  needs:
+    [
+      "kernel_matrix_testing_setup_env_security_agent_x64",
+      "upload_dependencies_security_agent_x64",
+    ]
+  timeout: 3h
+  variables:
+    ARCH: "x86_64"
+  parallel:
+    matrix:
+      - TAG:
+          [
+            "ubuntu_18.04",
+            "ubuntu_20.04",
+            "ubuntu_22.04",
+            "ubuntu_23.10",
+            "amzn_5.4",
+            "amzn_5.10",
+            "fedora_37",
+            "fedora_38",
+            "debian_10",
+            "debian_11",
+            "debian_12",
+            "centos_79",
+          ]
+        TEST_SET: [all_tests]
+
+kernel_matrix_testing_run_security_agent_tests_arm64:
+  extends:
+    - .kernel_matrix_testing_run_security_agent_tests
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  tags: ["arch:arm64"]
+  needs:
+    [
+      "kernel_matrix_testing_setup_env_security_agent_arm64",
+      "upload_dependencies_security_agent_arm64",
+    ]
+  timeout: 3h
+  variables:
+    ARCH: "arm64"
+  parallel:
+    matrix:
+      - TAG:
+          [
```

Comment:
Can we use the other array syntax for the `matrix` and `needs` fields? It reads easier.
```
- TAG:
    - ubuntu_18.04
    - ubuntu_20.04
    ...
```